app/Models/User.php

```diff
@@ -74,4 +77,55 @@ public function getCanAttribute()
     {
         return $this->getAllPermissions()->pluck('name');
     }
+
+    public function getUserNumberOfBookingRequestPerPeriod()
+    {
+        $numberOfBookingRequestPerPeriod = null;
+
+        foreach ($this->roles as $role) {
+            if ($role->number_of_bookings_per_period > $numberOfBookingRequestPerPeriod) {
+                $numberOfBookingRequestPerPeriod = $role->number_of_bookings_per_period;
+            }
+        }
+
+        return $numberOfBookingRequestPerPeriod;
+    }
+
+    public function getUserNumberOfDaysPerPeriod()
+    {
+        $numberOfDaysPerPeriod = null;
+
+        foreach ($this->roles as $role) {
+            if ($role->number_of_days_per_period > $numberOfDaysPerPeriod) {
+                $numberOfDaysPerPeriod = $role->number_of_days_per_period;
+            }
+        }
+
+        return $numberOfDaysPerPeriod;
+    }
+
+  public function canMakeAnotherBookingRequest($startDate): bool
+  {
+    $numberOfBookingRequestPerPeriod = $this->getUserNumberOfBookingRequestPerPeriod();
+    $numberOfDaysPerPeriod = $this->getUserNumberOfDaysPerPeriod();
+
+    if (is_null($numberOfBookingRequestPerPeriod) && is_null($numberOfDaysPerPeriod)) {
+      return true;
+    }
+
+    if (Carbon::parse($startDate)->diffInDays(Carbon::now()) > $numberOfDaysPerPeriod) {
+      return true;
+    }
+
+    $nbOfBookingRequest = BookingRequest::query()
+      ->whereHas('reservations', function (Builder $query) use ($numberOfDaysPerPeriod) {
+        $query->where('start_time', '>', Carbon::now());
+        $query->where('start_time', '<', Carbon::now()->addDays($numberOfDaysPerPeriod));
+      })
+      ->where('user_id', '=', $this->id)
+      ->where('status', 'like', '%approved%')
+      ->count();
```

Comment:
```suggestion
     $nbOfBookingRequest = Reservation::query()
    ->where('start_time', '>', Carbon::now())
    ->where('start_time', '<', Carbon::now()->addDays($numberOfDaysPerPeriod))
    ->where('user_id',  $this->id)
    ->where('status', 'not like', '%declined%')//status title pending
    ->join('booking_requests', 'booking_requests.id', '=', 'reservations.booking_request_id')
    ->select('reservations.*', 'booking_requests.user_id', 'booking_requests.status')
    ->count();
```
Here is possibility of how you can check the number of reservations. 
Just query the reservation model off the bat and add the two columns as joins for user_id and status.
Also i would check all statuses except declined since if an approval is "pending" the user would be able to make like100000 requests while reviews are waiting.

Reply:
I agree for the declined !